target/linux/realtek/files-5.10/drivers/clk/realtek/clk-rtl83xx.c

```diff
@@ -0,0 +1,750 @@
+
+#define OSC_RATE		25000000
+
+static const int rtcl_regs[SOC_COUNT][REG_COUNT][CLK_COUNT] = {
+	{
+		{ RTL838X_PLL_CPU_CTRL0, RTL838X_PLL_MEM_CTRL0, RTL838X_PLL_LXB_CTRL0 },
+		{ RTL838X_PLL_CPU_CTRL1, RTL838X_PLL_MEM_CTRL1, RTL838X_PLL_LXB_CTRL1 },
+	}, {
+		{ RTL839X_PLL_CPU_CTRL0, RTL839X_PLL_MEM_CTRL0, RTL839X_PLL_LXB_CTRL0 },
+		{ RTL839X_PLL_CPU_CTRL1, RTL839X_PLL_MEM_CTRL1, RTL839X_PLL_LXB_CTRL1 },
+	}
+};
+
+#define REG_SET(_rate, _ctrl0, _ctrl1)		\
+	{					\
+		.rate = _rate,			\
+		.ctrl0 = _ctrl0,		\
+		.ctrl1 = _ctrl1,		\
+	}
+
+struct rtcl_reg_set {
+	unsigned int rate;
+	unsigned int ctrl0;
+	unsigned int ctrl1;
+};
+
+static const struct rtcl_reg_set rtcl_838x_cpu_reg_set[] = {
+	REG_SET(300000000, 0x045c8, 0x1414530e),
+	REG_SET(325000000, 0x04648, 0x1414530e),
+	REG_SET(350000000, 0x046c8, 0x1414530e),
+	REG_SET(375000000, 0x04748, 0x1414530e),
+	REG_SET(400000000, 0x045c8, 0x0c14530e),
+	REG_SET(425000000, 0x04628, 0x0c14530e),
+	REG_SET(450000000, 0x04688, 0x0c14530e),
+	REG_SET(475000000, 0x046e8, 0x0c14530e),
+	REG_SET(500000000, 0x04748, 0x0c14530e),
+	REG_SET(525000000, 0x047a8, 0x0c14530e),
+	REG_SET(550000000, 0x04808, 0x0c14530e),
+	REG_SET(575000000, 0x04868, 0x0c14530e),
+	REG_SET(600000000, 0x048c8, 0x0c14530e),
+	REG_SET(625000000, 0x04928, 0x0c14530e)
+};
+
+static const struct rtcl_reg_set rtcl_838x_mem_reg_set[] = {
+	REG_SET(200000000, 0x041bc, 0x14018C80),
+	REG_SET(225000000, 0x0417c, 0x0c018C80),
+	REG_SET(250000000, 0x041ac, 0x0c018C80),
+	REG_SET(275000000, 0x0412c, 0x04018C80),
+	REG_SET(300000000, 0x0414c, 0x04018c80),
+	REG_SET(325000000, 0x0416c, 0x04018c80),
+	REG_SET(350000000, 0x0418c, 0x04018c80),
+	REG_SET(375000000, 0x041ac, 0x04018c80)
+};
+
+static const struct rtcl_reg_set rtcl_838x_lxb_reg_set[] = {
+	REG_SET(100000000, 0x043c8, 0x001ad30e),
+	REG_SET(125000000, 0x043c8, 0x001ad30e),
+	REG_SET(150000000, 0x04508, 0x1c1ad30e),
+	REG_SET(175000000, 0x04508, 0x1c1ad30e),
+	REG_SET(200000000, 0x047c8, 0x001ad30e)
+};
+
+static const struct rtcl_reg_set rtcl_839x_cpu_reg_set[] = {
+	REG_SET(400000000, 0x0414c, 0x00000005),
+	REG_SET(425000000, 0x04164, 0x00000005),
+	REG_SET(450000000, 0x0417c, 0x00000005),
+	REG_SET(475000000, 0x04194, 0x00000005),
+	REG_SET(500000000, 0x041ac, 0x00000005),
+	REG_SET(525000000, 0x041c4, 0x00000005),
+	REG_SET(550000000, 0x0412c, 0x00000004),
+	REG_SET(575000000, 0x0413c, 0x00000004),
+	REG_SET(600000000, 0x0414c, 0x00000004),
+	REG_SET(625000000, 0x0415c, 0x00000004),
+	REG_SET(650000000, 0x0416c, 0x00000004),
+	REG_SET(675000000, 0x0417c, 0x00000004),
+	REG_SET(700000000, 0x0418c, 0x00000004),
+	REG_SET(725000000, 0x0419c, 0x00000004),
+	REG_SET(750000000, 0x041ac, 0x00000004),
+	REG_SET(775000000, 0x041bc, 0x00000004),
+	REG_SET(800000000, 0x041cc, 0x00000004),
+	REG_SET(825000000, 0x041dc, 0x00000004),
+	REG_SET(850000000, 0x041ec, 0x00000004)
+};
+
+static const struct rtcl_reg_set rtcl_839x_mem_reg_set[] = {
+	REG_SET(100000000, 0x041CC, 0x00000000),
```

Comment:
stay consistent with your capitlization of your hexes; lower-case is generally preferred)